### ISSUE TITLE
Как правильно использовать gulp-bem?

### ISSUE THREAD

**Issue by SmirnovW** [bem/bem-forum-content-ru#617]
Всем привет! Осваиваю методологию BEM и хочу внедрить её в наши проекты. Мы используем gulp и не хотим от него отказываться, поэтому решил попробовать gulp-bem, но столкнулся с не понимаем того как работаем этот модуль.
На гите есть пример:
var gulp    = require('gulp');
var bem     = require('gulp-bem');
var concat  = require('gulp-concat');
 
var levels = ['base', 'blocks'];
var tree = bem(levels);
 
tree.deps('blocks/page')
    .pipe(bem.src('{bem}.css'))
    .pipe(concat('index.css'))
    .pipe(gulp.dest('./dist'))

Вот тут всё понятно:
var levels = ['base', 'blocks'];
var tree = bem(levels);

А вот тут возникла сложность:
tree.deps('blocks/page')
Что делает этот код? Спасибо!

**Comment by belozyorcev** [comment]
Этот участок кода ищет на всех уровнях переопределения необходимые и зависимые блоки, которые затем попадают в сборку.

**Comment by SmirnovW** [comment]
Т.е. например у меня такая структура проекта:
blocks/
       header/
              header.css
              header__content.css              
              header.js
Этот код найдёт эти блоки?

**Comment by belozyorcev** [comment]
Да, если они используются в проекте.
Только не блоки, а технологии. У тебя ```header``` - это блок.

Или у тебя есть блоки: ```footer```, ```page```, ```header```, ```menu```
но используются только ```header``` и ```menu``` (то в сборке будут только ```header``` и ```menu```).

А если, например, у тебя в зависимости ```header``` есть блок ```page```
то результате конечной сборки будут блоки ```page```, ```header```, ```menu```.

При этом произойдёт склеивание всех версий блоков из разных уровней.
Например на первом уровне блок ```header``` имел размеры
```width: 200px; height: 200px```
на втором уровне он изменил высоту: ```height: 192px```
на третьем уровне он поменял цвет: ```background-color: #844```

в результат мы получили нужный нам header.
```
width: 200px; 
height: 192px;
background-color: #844;
```

**Comment by SmirnovW** [comment]
Так, а как я должен указать какие блоки я использую? Там в примере вроде в ничего такого не указывается, или я что то не до конца понял?

**Comment by belozyorcev** [comment]
Ты через bemjson делаешь разметку? Сборщик сам определяет какие блоки задействованы исходя из разметки.

А ещё почитай про deps
https://ru.bem.info/technology/deps/about/

**Comment by SmirnovW** [comment]
Я создал index.bemjson.js, добавил в него простой код:
{ block: 'menu' }
Положил этот файл тут "blocks/page"
По пути "blocks/menu" лежит файл menu.css

В gulpfile.js написал следующее:
var gulp = require('gulp'),
	bem = require('gulp-bem'),
	concat = require('gulp-concat');

var levels = ['blocks'];
var tree = bem(levels);

gulp.task('default', function () {
	tree.deps('blocks/page')
	.pipe(bem.src('{bem}.css'))
	.pipe(concat('index.css'))
	.pipe(gulp.dest('./dist'));
});

Но почему то ничего не создаётся. Что ещё я упустил?

**Comment by apsavin** [comment]
Не пользовался gulp-bem, но предположу, что проблема в том, что файл должен иметь имя page, а не index.

**Comment by SmirnovW** [comment]
Попробовал поменять на page, ничего не изменилось. Пойду писать автору.

**Comment by belozyorcev** [comment]
@SmirnovW посмотри здесь пример.
https://github.com/iamstarkov/gulp-bem-stub
